neuralprophet/time_net.py

```diff
@@ -832,26 +773,24 @@ def compute_components(self, inputs, meta):
                 )
         if self.config_regressors is not None and "regressors" in inputs:
             if "additive" in inputs["regressors"].keys():
-                components["future_regressors_additive"] = self.scalar_features_effects(
-                    features=inputs["regressors"]["additive"], params=self.regressor_params["additive"]
+                components["future_regressors_additive"] = self.future_regressors(
+                    inputs["regressors"]["additive"], "additive"
                 )
             if "multiplicative" in inputs["regressors"].keys():
-                components["future_regressors_multiplicative"] = self.scalar_features_effects(
-                    features=inputs["regressors"]["multiplicative"], params=self.regressor_params["multiplicative"]
+                components["future_regressors_multiplicative"] = self.future_regressors(
+                    inputs["regressors"]["multiplicative"], "multiplicative"
                 )
-            for regressor, configs in self.regressors_dims.items():
+            for regressor, configs in self.future_regressors.regressors_dims.items():
                 mode = configs["mode"]
                 index = []
                 index.append(configs["regressor_index"])
                 if mode == "additive":
                     features = inputs["regressors"]["additive"]
-                    params = self.regressor_params["additive"]
+                    mode = "additive"
                 else:
                     features = inputs["regressors"]["multiplicative"]
-                    params = self.regressor_params["multiplicative"]
-                components[f"future_regressor_{regressor}"] = self.scalar_features_effects(
-                    features=features, params=params, indices=index
-                )
```

Comment:
What exactly is this section of the new code doing? The following looks kind of redundant
```python
if mode == "additive":
    ...
    mode = "additive"
```

Reply:
@alfonsogarciadecorral can you have a quick look at this?